azkaban-common/src/main/java/azkaban/executor/dispatcher/ExecutorDispatcher.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2015 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.executor.dispatcher;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+
+/** Implementation of the ExecutorDispatcher.
+ *  @param K executor object type.
+ *  @param V dispatching object type.
+ * */
+public class ExecutorDispatcher<K,V> implements Dispatcher<K, V> {
+  private static Logger logger = Logger.getLogger(CandidateComparator.class);
+
+  private CandidateFilter<K,V> filter;
+  private CandidateComparator<K> comparator;
+
+  // getter of the filter.
+  public CandidateFilter<K,V> getFilter(){
+    return this.filter;
+  }
+
+  // setter of the filer.
+  public CandidateFilter<K,V> setFilter(CandidateFilter<K,V> value){
+    CandidateFilter<K,V> returnVal = this.filter;
+    this.filter = value;
+    return returnVal;
+  }
+
+  // getter of the comparator.
+  public CandidateComparator<K> getComparator(){
+    return this.comparator;
+  }
+
+  // setter of the comparator.
+  public CandidateComparator<K> setComparator(CandidateComparator<K>  value){
+    CandidateComparator<K> returnVal = this.comparator;
+    this.comparator = value;
+    return returnVal;
+  }
+
+  /**constructor of the class.
+   * @param filter CandidateFilter object to be used to perform the candidate filtering.
+   * @param comparator CandidateComparator object to be used to find the best suit candidate from the filtered list.
+   * */
+  public ExecutorDispatcher(CandidateFilter<K,V> filter,
+      CandidateComparator<K> comparator){
+    this.filter = filter;
+    this.comparator = comparator;
+  }
+
+  @Override
+  public K getNext(List<K> candidateList, V dispatchingObject) {
```

Comment:
should this method be called getMax, getBest, getTop, or something like that?
getNext gives me the impression of an iterator, so I would expect two calls to getNext() return 2 different values.